File: C/pset2/problems-2020-x-readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    float l = 0;
    // Starting First Word
    float w = 1;
    float s = 0;
    // Iterate until \0
    for (int i = 0; text[i] != '\0'; i++)
    {
        // If alphabet count 1 to l
        if (isalpha(text[i]) != 0)
        {
            l++;
        }
        // Word Count
        if (isblank(text[i]) == 0 && isblank(text[i - 1]) != 0)
        {
            w++;
        }
        // Sentences Count
        if (text [i] == '.' || text [i] == '?' || text [i] == '!')
        {
            s++;
        }

    }

    // Calculate Grade Using Float
    float letters = (l / w) * 100;
    float sentences = (s / w) * 100;
    float grade = 0.0588 * letters - 0.296 * sentences - 15.8;

    //Make Printf
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %.0f\n", round(grade));
    }
}

File: Python, SQL, HTML, CSS, Flask/pset6/problems-2020-x-sentimental-readability/readability.py
from cs50 import get_string

# Asking for input from user
text = get_string("Text: ")

# Defining a global variable to count letter, word and sentence
letter = 0
word = 1
sentence = 0

# Examine each character of the string
for i in range(len(text)):
    if text[i].isalpha() == True:
        letter += 1
    if text[i].isspace() == True and text[i+1].isspace() == False:
        word += 1
    if text[i] == "." or text[i] == "?" or text[i] == "!":
        sentence += 1

# Calculating grade
grade = round(0.0588 * (letter/word*100) - 0.296 * (sentence/word*100) - 15.8)

# Printing the result
if grade >= 16:
    print("Grade 16+")
elif grade <= 1:
    print("Before Grade 1")
else:
    print("Grade " + str(grade))